### DISCUSSION TITLE
Creating a wrapper for Unity

### DISCUSSION THREAD

**Discussion by Kulgann** [zeux/meshoptimizer#474] [General]
I would like to use this lib in my unity projects, since it seems to be exactly what I need. How would I go about to creating the wrapper?
As far as I know I need C style functions to call from the dll on the c# side in Unity, but I cant seem to find them in the source.
Any help would be greatly appreciated.

**Comment by zeux** [comment]
You might be interested in https://github.com/maxilevi/MeshOptimizer.NET - it was made against an earlier version of meshoptimizer but the same technique should apply.

**Comment by xPaw** [comment]
If anyone is interested, I have a C# port of the decoders here:

https://github.com/ValveResourceFormat/ValveResourceFormat/blob/master/ValveResourceFormat/Compression/MeshOptimizerIndexDecoder.cs
https://github.com/ValveResourceFormat/ValveResourceFormat/blob/master/ValveResourceFormat/Compression/MeshOptimizerVertexDecoder.cs
https://github.com/ValveResourceFormat/ValveResourceFormat/blob/master/ValveResourceFormat/Compression/MeshOptimizerVertexDecoder.Simd.cs
